app/models/invalid_identities.rb

```diff
@@ -0,0 +1,22 @@
+class InvalidIdentities
+
+  def initialize(identities, progress_bar)
+    @identities = identities
+    @progress_bar = progress_bar
+  end
+
+  def remove_from_db
+    @identities.each do |identity|
+      next if !identity.last_sign_in_at.nil?
+      next if identity.project_roles.present?
+      next if identity.catalog_managers.present?
+      next if identity.clinical_providers.present?
+      next if identity.service_providers.present?
+      next if identity.super_users.present?
+      identity.destroy
+      @progress_bar.increment
```

Comment:
I'm assuming you're intending this class to be possibly reused. In that case, I would pull out the `@progress_bar.increment` and replace it with something like `yield(identity) if block_given?`. Now we can do:
```invalid_identities.remove_from_db { |identity| progress_bar.increment }```
and maybe down the road for another application:
```
invalid_identities.remove_from_db do |identity|
  email_goodbye(identity)
  progress_bar.increment
end
```
Or just `invalid_identities.remove_from_db` if you're not feeling fancy.
My gripe really if that the progress bar seem `puts`y or `console.log`y and shouldn't be in a utility class.